flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/FanOutRecordPublisherConfiguration.java

```diff
@@ -0,0 +1,438 @@
+		return deregisterStreamBaseBackoffMillis;
+	}
+
+	/**
+	 * Get maximum backoff millis for the deregister stream operation.
+	 */
+	public long getDeregisterStreamMaxBackoffMillis() {
+		return deregisterStreamMaxBackoffMillis;
+	}
+
+	/**
+	 * Get exponential backoff power constant for the deregister stream operation.
+	 */
+	public double getDeregisterStreamExpConstant() {
+		return deregisterStreamExpConstant;
+	}
+
+	/**
+	 * Get maximum retry attempts for the register stream operation.
+	 */
+	public int getDeregisterStreamMaxRetries() {
+		return deregisterStreamMaxRetries;
+	}
+
+	// ------------------------------------------------------------------------
+	//  describeStream() related performance settings
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Get maximum retry attempts for the describe stream operation.
+	 */
+	public int getDescribeStreamMaxRetries() {
+		return describeStreamMaxRetries;
+	}
+
+	/**
+	 * Get base backoff millis for the describe stream operation.
+	 */
+	public long getDescribeStreamBaseBackoffMillis() {
+		return describeStreamBaseBackoffMillis;
+	}
+
+	/**
+	 * Get maximum backoff millis for the describe stream operation.
+	 */
+	public long getDescribeStreamMaxBackoffMillis() {
+		return describeStreamMaxBackoffMillis;
+	}
+
+	/**
+	 * Get exponential backoff power constant for the describe stream operation.
+	 */
+	public double getDescribeStreamExpConstant() {
+		return describeStreamExpConstant;
+	}
+
+	// ------------------------------------------------------------------------
+	//  describeStreamConsumer() related performance settings
+	// ------------------------------------------------------------------------
+
+	/**
+	 * Get maximum retry attempts for the describe stream operation.
+	 */
+	public int getDescribeStreamConsumerMaxRetries() {
+		return describeStreamConsumerMaxRetries;
+	}
+
+	/**
+	 * Get base backoff millis for the describe stream operation.
+	 */
+	public long getDescribeStreamConsumerBaseBackoffMillis() {
+		return describeStreamConsumerBaseBackoffMillis;
+	}
+
+	/**
+	 * Get maximum backoff millis for the describe stream operation.
+	 */
+	public long getDescribeStreamConsumerMaxBackoffMillis() {
+		return describeStreamConsumerMaxBackoffMillis;
+	}
+
+	/**
+	 * Get exponential backoff power constant for the describe stream operation.
+	 */
+	public double getDescribeStreamConsumerExpConstant() {
+		return describeStreamConsumerExpConstant;
+	}
+
+	/**
+	 * Get efo registration type.
+	 */
+	public EFORegistrationType getEfoRegistrationType() {
+		return efoRegistrationType;
+	}
+
+	/**
+	 * Get consumer name, will be null if efo registration type is 'NONE'.
+	 */
+	public Optional<String> getConsumerName() {
+		return Optional.ofNullable(consumerName);
+	}
+
+	/**
+	 * Get stream consumer arns, will be null if efo registration type is 'LAZY' or 'EAGER'.
+	 */
+	public Optional<Map<String, String>> getStreamConsumerArns() {
```

Comment:
@xiaolong-sn Usually you would just return an empty `Map` rather than an `Optional`. This method is not actually required, and I have deleted it in one of my subsequent pull requests. Feel free to delete it from here though.

Reply:
I've deleted this method. But I think there is  some difference between `Optional.empty`  and an empty map.